src/adventure/lineparsers.cpp

```diff
@@ -0,0 +1,99 @@
+#include "lineparsers.h"
+
+AbstractStatefulLineParser::~AbstractStatefulLineParser() {}
+
+QString AbstractStatefulLineParser::getLastSuccessVal()
+{
+    return m_lastSuccessVal;
+}
+
+bool AccomplishedTaskParser::parse(QString line)
+{
+    // REVISIT: there are at least three different versions of this
+    //   accomplished
+    //   knowledgeable (xp?)
+    //   travelled (tp?)
+    return line.indexOf("With the task complete, you feel more") == 0;
```

Comment:
This is technically not super efficient since it checks the whole string for when this begins. The same is probably true for other captures here down the road. A regular expression would know to just check the beginning and abort early (or https://doc.qt.io/qt-5/qstring.html#startsWith with less complexity).

btw, I'm a huge fan of using https://www.regexpal.com/ as a tool to help build regular expressions and test them. We could even do some unit tests to do sanity checks.

Reply:
Yeah, d'oh. I guess subconsciously I was thinking that this _was_ `startsWith`, but of course it's not (it's `checkEverywhereUnnecessarily`), will fix.

Reply:
fixed throughout